dex/tests/close-markets/programs/close-markets/src/lib.rs

```diff
@@ -0,0 +1,338 @@
+            &[&ctx.accounts.prune_auth.signer_seeds()],
+        )?;
+
+        Ok(())
+    }
+
+    pub fn close_market(ctx: Context<CloseMarket>) -> ProgramResult {
+        let ix = serum_dex::instruction::close_market(
+            &ctx.accounts.dex_program.key(),
+            &ctx.accounts.serum_market.key(),
+            &ctx.accounts.request_queue.key(),
+            &ctx.accounts.event_queue.key(),
+            &ctx.accounts.bids.key(),
+            &ctx.accounts.asks.key(),
+            &ctx.accounts.prune_auth.key(),
+            &ctx.accounts.payer.key(),
+        )?;
+
+        let seeds = &[b"prune_auth".as_ref(), &[ctx.accounts.prune_auth.bumps.prune_auth]];
+        let signer = &[&seeds[..]];
+
+        solana_program::program::invoke_signed(
+            &ix,
+            &ToAccountInfos::to_account_infos(ctx.accounts),
+            signer,
+        )?;
+
+        Ok(())
+    }
+}
+
+#[derive(Accounts)]
+#[instruction(bumps: Bumps)]
+pub struct InitializeMarket<'info> {
+    #[account(mut)]
+    pub payer: Signer<'info>,
+    #[account(
+        init,
+        seeds = [b"prune_auth".as_ref()],
+        bump = bumps.prune_auth,
+        payer = payer
+    )]
+    pub prune_auth: Box<Account<'info, PruneAuth>>,
+    #[account(
+        init,
+        mint::decimals = 6,
+        mint::authority = payer,
+        seeds = [b"usdc_mint".as_ref()],
+        bump = bumps.usdc_mint,
+        payer = payer
+    )]
+    pub usdc_mint: Box<Account<'info, Mint>>,
+    #[account(
+        init,
+        mint::decimals = 6,
+        mint::authority = payer,
+        seeds = [b"serum_mint".as_ref()],
+        bump = bumps.serum_mint,
+        payer = payer
+    )]
+    pub serum_mint: Box<Account<'info, Mint>>,
+    #[account(
+        init,
+        seeds = [SERUM_MARKET_SEED.as_ref()],
+        bump = bumps.serum_market,
+        space = SERUM_MARKET_SPACE, 
+        payer = payer,
+        owner = dex_program.key(
+    ))]
+    pub serum_market: UncheckedAccount<'info>,
+    #[account(
+        init,
+        seeds = [REQUEST_QUEUE_SEED.as_ref()],
+        bump = bumps.request_queue,
+        space = REQUEST_QUEUE_SPACE,
+        payer = payer,
+        owner = dex_program.key(
+    ))]
+    pub request_queue: UncheckedAccount<'info>,
+    #[account(
+        init,
+        token::mint = serum_mint,
+        token::authority = vault_signer,
+        seeds = [COIN_VAULT_SEED.as_ref()],
+        bump = bumps.coin_vault,
+        payer = payer
+    )]
+    pub coin_vault: Box<Account<'info, TokenAccount>>,
+    #[account(
+        init,
+        token::mint = usdc_mint,
+        token::authority = vault_signer,
+        seeds = [PC_VAULT_SEED.as_ref()],
+        bump = bumps.pc_vault,
+        payer = payer
+    )]
+    pub pc_vault: Box<Account<'info, TokenAccount>>,
+    // TODO probably no way to verify these seeds since it uses a different heuristic
+    pub vault_signer: UncheckedAccount<'info>,
+    // These shouldn't need to be signers
```

Comment:
Technically its better if they are indeed signers right? Since then you dont have to worry about unitialized accounts being hijacked. But it doesnt matter for the test.

Reply:
good idea